src/intersection-observer-polyfill.js

```diff
@@ -273,6 +286,19 @@ export class IntersectionObserverPolyfill {
       }
     }
 
+
+    // Add a mutation observer to tick ourself
+    if (!this.mutationObserver_) {
+      this.viewport_ = Services.viewportForDoc(element);
+      this.mutationObserver_ = new MutationObserver(
+          this.handleMutationObserverNotification_.bind(this)
+      );
+      this.mutationObserver_.observe(element.ownerDocument, {
```

Comment:
Use the element's ampdoc's document, not the element's `ownerDocument`.

Reply:
ampdoc.getRootNode can give ShadowRoot. is it the right tree to listen for mutation events?

i.e., can a change from a sibling node of this ShadowRoot affect its visibility?